Firestore/core/src/firebase/firestore/util/comparison.h

```diff
@@ -180,6 +180,28 @@ bool DoubleBitwiseEquals(double left, double right);
  */
 size_t DoubleBitwiseHash(double d);
 
+template <typename T>
+class Equatable {
+ public:
+  friend bool operator!=(const T& lhs, const T& rhs) {
+    return !(lhs == rhs);
+  }
+};
+
+template <typename T>
+class Comparable : public Equatable<T> {
+ public:
+  friend bool operator>(const T& lhs, const T& rhs) {
+    return rhs < lhs;
+  }
+  friend bool operator<=(const T& lhs, const T& rhs) {
+    return !(rhs < lhs);
+  }
+  friend bool operator<=(const T& lhs, const T& rhs) {
```

Comment:
I think you mean operator>=

Reply:
I did. That's what I get for not actually compiling this code :-). Also added tests.